File: monkeys.rb
alpha = ("a".."z").to_a

unfinished = true
best_guess = "a"
counter = 0

puts
print best_guess

while (unfinished)

	streak = 0
	guess = alpha.sample 

	if (guess == alpha[0])
		on_streak = true

		while (on_streak) 
			if guess == alpha[streak]
				if (guess > best_guess)
					best_guess = guess
					print guess
				end
				streak = streak + 1 
				if (guess == "z")
					puts ; puts "amazing!"
					exit
				end
				guess = alpha.sample
			else
				on_streak = false
			end
		end

	end

end 

File: monkeys.js
var str = "abcdefghijklmnopqrstuvwxyz"
var alpha = str.split("");

var unfinished = true;
var best_guess = "a";
var counter = 0;

console.log(best_guess);

while (unfinished){
	var streak = 0;
	var guess = alpha[Math.floor(Math.random() * alpha.length)];

	if (guess === alpha[0]){
		var on_streak = true;

		while (on_streak){

			if (guess === alpha[streak]){

				if (guess > best_guess){
					best_guess = guess;
					console.log(guess);
				};

				streak++;

				if (guess === "z"){ console.log("amazing!");};

				guess = alpha[Math.floor(Math.random() * alpha.length)];

			} else {
				on_streak = false;

			};
		};
	};
}; // end while
